protocol-units/da-sequencer/node/src/tests/client.rs

```diff
@@ -219,15 +226,203 @@ async fn test_produce_block_and_stream() {
+	});
+
+	let _ = tokio::time::sleep(tokio::time::Duration::from_millis(300)).await;
+
+	let connection_url = Url::parse(&format!("http://{}", grpc_address)).unwrap();
+	let mut client = GrpcDaSequencerClient::try_connect(&connection_url)
+		.await
+		.expect("Failed to connect");
+
+	let tx = Transaction::test_only_new(b"abc".to_vec(), 1, 123);
+	let txs = FullNodeTxs::new(vec![tx]);
+	let batch_bytes = bcs::to_bytes(&txs).unwrap();
+	let signature = signing_key.sign(&batch_bytes);
+	let signature = SigningSignature::try_from(&signature.to_bytes()[..]).unwrap();
+	let data = serialize_full_node_batch(verifying_key, signature, batch_bytes);
+
+	let res = client.batch_write(BatchWriteRequest { data }).await;
+	let response = res.expect("batch_write failed");
+	assert!(response.answer);
+
+	grpc_task.abort();
+	let _ = grpc_task.await;
+
+	main_loop.abort();
+	let _ = main_loop.await;
+}
+
+/// Submit a batch with an empty the white list.
+/// The batch is rejected because the submitter public key is not part of the white list.
+#[tokio::test]
+async fn test_grpc_client_should_write_one_batch_with_an_empty_whitelist() {
+	let (request_tx, request_rx) = mpsc::channel(100);
+
+	let config = DaSequencerConfig::default();
+	let signing_key = generate_signing_key();
+	let verifying_key = signing_key.verifying_key();
+
+	let whitelist = make_test_whitelist(vec![]);
+
+	// Start gprc server. Define a different address for each test.
+	let grpc_address = "0.0.0.0:30704"
+		.parse::<SocketAddr>()
+		.expect("Bad da sequencer listener address.");
+	let _grpc_jh =
+		tokio::spawn(async move { run_server(grpc_address, request_tx, whitelist).await });
+
+	let storage_mock = StorageMock::new();
+	let celestia_mock = CelestiaMock::new();
+	let _loop_jh = tokio::spawn(run(config, request_rx, storage_mock, celestia_mock));
+
+	tokio::time::sleep(tokio::time::Duration::from_millis(300)).await;
+
+	let tx = Transaction::test_only_new(b"test data".to_vec(), 1, 123);
+	let txs = FullNodeTxs::new(vec![tx]);
+	let batch_bytes = bcs::to_bytes(&txs).expect("Serialization failed");
+	let signature = signing_key.sign(&batch_bytes);
+	let signature = SigningSignature::try_from(&signature.to_bytes()[..]).unwrap();
+
+	let serialized = serialize_full_node_batch(verifying_key, signature, batch_bytes);
+
+	let connection_url = Url::parse(&format!("http://{}", grpc_address)).unwrap();
+	let mut client = GrpcDaSequencerClient::try_connect(&connection_url)
+		.await
+		.expect("gRPC client connection failed");
+
+	let request = BatchWriteRequest { data: serialized };
+	let res = client.batch_write(request).await.expect("Failed to send the batch.");
+	tracing::info!("{res:?}");
+	assert!(!res.answer);
+
+	tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
+}
+
+/// Submit a batch and sign it with a different key than the one declared in the whitelist.
+/// The batch is rejected because the submitter public key is not part of the white list.
```

Comment:
```suggestion
// Submit a batch and sign it with a different key than the one declared in the whitelist.
// The batch is rejected because the submitter public key is not part of the white list.
```
The extra comments here are helpful and a nice idea. However their not doc comments just general comments